### PR TITLE
Attribute based access control implementation

### PR DESCRIPTION
**Pull request by litvinovg**
**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues/3027)**

[Vitro PR](https://github.com/vivo-project/Vitro/pull/398)

# What does this pull request do?
This pull request is a VIVO part of new authorization mechanism that provides flexible way to configure access rules and check access requests.

# What's new?
This pull requests contains:
* Datasets for entity policies in n3 files.
* Java policies were removed.
* Startup listeners file was modified in accordance with linked Vitro PR
* Authorization requests in generator were aligned with Vitro PR
You can find more information in Vitro PR.

# How should this be tested?
There are 2 ways to test it:
* Migration from currently in use VIVO instance
Apply changes in PR for Vitro and for VIVO, build and deploy your VIVO.
Check if access works the same as it worked before for object properties, data properties, faux object properties and faux data properties. 
Check if access works the same as it worked in ARM. Standard VIVO checks are required to test this PR.
If you want to try new policies, try edit policies in firsttime directory, reload VIVO and see results.
* Migration from Advanced role management
Apply changes in PR for Vitro and for VIVO
Make sure to retain ARM entity permission configurations in auth/firsttime folder to retain auth graph state for conversion of ARM permissions into policy datasets.
Build and deploy your VIVO.
Check if access works the same as it worked in ARM.
If you want to try new policies, try edit policies in firsttime directory, reload VIVO and see results.

# Additional notes
* Documentation will need to be updated.

# Interested parties
@chenejac @vivo-project/vivo-committers

### CODE REVIEW

**Review by chenejac — CHANGES_REQUESTED** [review]
Just one small comment here, which is more likely that explanation is needed than fixing.

File: api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java
Comment (litvinovg) [reply api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java]: Fixed. Thanks for catching that.

**Review by chenejac — DISMISSED** [review]
@litvinovg great work. Let's wait for one more review, especially review of the ABAC ontology.

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/generators/ManageLabelsForPersonGenerator.java
Comment (ivanmrsulja) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/generators/ManageLabelsForPersonGenerator.java:337]: I would change variable names to make sense with the new type (eg. `dpsAccessObject`).

File: api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java
Comment (ivanmrsulja) [inline api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java:301]: Same thing here.

**Review by ivanmrsulja — COMMENTED** [review]
LGTM! I have left a few comments discussing minor variable naming changes

File: api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java
Comment (litvinovg) [reply api/src/main/java/org/vivoweb/webapp/controller/freemarker/CreateAndLinkResourceController.java]: Done

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/generators/ManageLabelsForPersonGenerator.java
Comment (litvinovg) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/generators/ManageLabelsForPersonGenerator.java]: Done

**Comment by matthiasluehr** [comment]
Tested migration of existing custom roles and related permissions. Looks good.

**Review by chenejac — APPROVED** [review]
@litvinovg VIVO PR is completed from my point of view.

**Review by ivanmrsulja — APPROVED** [review]
Everything looks fine from my point of view. Good job!

**Review by chenejac — APPROVED** [review]
@litvinovg well done

**Review by chenejac — APPROVED** [review]

**Pull request merged**
